### PR TITLE
[CHI-XXX] adding new attribute variant for chi-button.

### PR DESCRIPTION
**Pull request by srinath-koneru**
Adding the new attributes to house both outline and flat properties of a button.
Deleted the old individual outline and flat props.

### CODE REVIEW

**Comment by srinath-koneru** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/78/ for more details.

**Comment by srinath-koneru** [comment]
@dani-cl-madrid @mattnickles  can you please review this, I have updated the chi-button with variant as prop and values are outline, flat or '' . TIA

File: src/website/views/custom-elements/button.pug [hunk 1]
```diff
@@ -36,57 +36,57 @@ p.-text
...
+  | To remove a buttons solid background and keep it's colored border, set the attribute
+  | <code>variant</code> value to <code>outline</code>.
 .example.-mb--3
   .a-grid.-noGutter
     .a-col.-w--12.-w-md--6
       .-p--3.-d--flex(style="justify-content: center;")
         each type, val in {Primary:'primary', Dark:'dark'}
           div.-pr--2
-            chi-button(color=type, outline)= val
+            chi-button(color=type, variant="outline")= val
```
Comment (jllr) [inline src/website/views/custom-elements/button.pug]: @srinath-koneru js attributes should be followed by single quotes and html attributes should have double quotes in order to follow best practices. Please change it in these files.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/87/ for more details.

**Comment by srinath-koneru** [comment]
@mattnickles this PR can be merged.

File: src/website/views/custom-elements/button.pug [hunk 2]
```diff
@@ -116,16 +116,16 @@ p.-text
   .-p--3.-d--flex
     chi-button(type='icon')
       chi-icon(icon='atom')
-    chi-button(flat, class='-ml--2', type='icon')
+    chi-button(variant='flat', class='-ml--2', type='icon')
```
Comment (dani-cl-madrid) [inline src/website/views/custom-elements/button.pug]:
```suggestion
    chi-button.-ml--2(variant='flat, type='icon')
```
@srinath-koneru This is the preferred way to include classes in Pug. Use the `class=` notation when using variables.

Comment (srinath-koneru) [reply src/website/views/custom-elements/button.pug]: Alright.

**Comment by srinath-koneru** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/93/ for more details.

**Comment by srinath-koneru** [comment]
@mattnickles can you please merge this PR.

**Pull request merged**